src/pykeen/models/unimodal/hake.py

```diff
@@ -0,0 +1,173 @@
+# coding=utf-8
+"""Implementation of the HAKE model."""
+
+from typing import Optional
+
+import torch
+from torch import nn
+
+from pykeen.losses import Loss
+from pykeen.models import EntityRelationEmbeddingModel
+from pykeen.regularizers import Regularizer
+from pykeen.triples import TriplesFactory
+from pykeen.utils import get_embedding, get_embedding_in_long_canonical_shape
+
+
+class HAKE(EntityRelationEmbeddingModel):
+    """An implementation of HAKE [zhang2020]_.
+
+    .. seealso ::
+        Official implementation: https://github.com/MIRALab-USTC/KGE-HAKE
+
+    """
+
+    # There are no value ranges for HPO given in the paper
+    # These values are the values ranges used in the usage examples from
+    # https://github.com/MIRALab-USTC/KGE-HAKE/blob/31049c2a00492533693699636d9c0a3a2b299fc8/README.md
+    hpo_default = dict(
+        embedding_dim=dict(type=int, low=256, high=512, q=32),
+        gamma=dict(type=float, low=6.0, high=24.0),
+        modulus_weight=dict(type=float, low=0.5, high=4.0),
+        phase_weight=dict(type=float, low=0.5, high=1.0),
+    )
+
+    def __init__(
+        self,
+        triples_factory: TriplesFactory,
+        embedding_dim: int = 500,
+        gamma: float = 12.0,
+        epsilon: float = 2.0,
+        modulus_weight: float = 1.0,
+        phase_weight: float = 0.5,
+        loss: Optional[Loss] = None,
+        predict_with_sigmoid: bool = False,
+        automatic_memory_optimization: Optional[bool] = None,
+        preferred_device: Optional[str] = None,
+        random_seed: Optional[int] = None,
+        regularizer: Optional[Regularizer] = None,
+    ):
```

Comment:
please add docstring to `__init__()` that has explanations of the gamma, epsilon, modulus_weight, and phase_weight parameters.

Reply:
still important